provider/kubernetes/kubernetes.go

```diff
@@ -190,6 +190,16 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 		}
 		templateObjects.TLS = append(templateObjects.TLS, tlsSection...)
 
+		var weightAllocator weightAllocator = &defaultWeightAllocator{}
+		annotationPercentageWeights := getAnnotationName(i.Annotations, annotationKubernetesServiceWeights)
+		if _, ok := i.Annotations[annotationPercentageWeights]; ok {
+			fractionalAllocator, err := newFractionalWeightAllocator(i, k8sClient)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create fractional weight allocator for ingress %s/%s: %v", i.Namespace, i.Name, err)
```

Comment:
Sorry, I probably should have been more clear in my original comment: what I meant by "failing fast" was that we skip the affected Ingress. Returning an error would have the negative effect that a single "bad" Ingress would cause processing for all other "good" Ingresses to stop as well, which is likely not what we want.

Reply:
@timoreimann NP. Updated :)